### PR TITLE
anoblet/feat/vaadin-icon

### PR DESCRIPTION
**Pull request by anoblet**
This is a large PR with plenty of room for error. I've tested as much as I can, though please review.

The most notable change was the use of `getIconNames` which isn't present in `vaadin-icon` and had to be brought into `icon-preview.js`

---

https://github.com/conversionxl/aybolit/blob/a5d7ec9bc154c1e314272386af96ae9ef0baa122/packages/storybook/cxl-lumo-styles/icons-preview.js#L109

https://github.com/PolymerElements/iron-iconset-svg/blob/c575f7f30be3174607a76ccf15c16ce4c2416f4c/iron-iconset-svg.js#L102

https://github.com/vaadin/web-components/blob/master/packages/icon/src/vaadin-iconset.js

### CODE REVIEW

**Review by pawelkmpt — COMMENTED** [review]
```diff
- feat(cxl-ui): migrate iron-icon to vaadin-icon
+ feat(cxl-ui): migrate to vaadin-icon
```

**Comment by kylebrodeur** [comment]
Task linked: [CU-344efq1 Update / Replace <iron-icon> with <vaadin-icon>](https://app.clickup.com/t/344efq1)

**Review by pawelkmpt — APPROVED** [review]
Icons load on Storybook so it seems to be fine, but let's wait for @saas786 review as he knows more JS details 🙂

**Comment by saas786** [comment]
@anoblet 

did you explore the possibility of removing `iron-icon` bloat (as its no longer a deps), from packages?

I can see it:

```
λ npm ls @polymer/iron-icon
aybolit-workspace@ D:\source\repos\github\cxl\aybolit
├─┬ @conversionxl/cxl-lumo-styles@1.2.0 -> .\packages\cxl-lumo-styles
│ └─┬ @vaadin/vaadin-lumo-styles@23.2.8
│   └── @polymer/iron-icon@3.0.1
└─┬ @conversionxl/cxl-ui@1.2.0 -> .\packages\cxl-ui
  └─┬ @cwmr/iron-star-rating@3.1.4 invalid: "github:conversionxl/iron-star-rating" from packages/cxl-ui
    ├── @polymer/iron-icon@3.0.1 deduped
    └─┬ @polymer/iron-icons@3.0.1
      └── @polymer/iron-icon@3.0.1 deduped
```

ref:
- https://github.com/vaadin/web-components/blob/v23.2.8/packages/vaadin-lumo-styles/package.json#L42

We can defer it now, and remove it later (as seems like **v24** will remove it):

ref:
- https://github.com/vaadin/web-components/releases/tag/v24.0.0-alpha5
- https://github.com/vaadin/web-components/blob/v24.0.0-alpha5/packages/vaadin-lumo-styles/package.json#L42
- https://github.com/vaadin/web-components/pull/5010

---

Why we still have `@cwmr/iron-star-rating` as deps? Since https://github.com/conversionxl/aybolit/pull/228 we now have our own custom star icon?

As it also bring in `@polymer/iron-icon`.

ref:
- https://github.com/conversionxl/aybolit/blob/4c9f0cc5f61a8f4caa6d639169388a7cc488398c/packages/cxl-ui/package.json#L15

---

I think we should wait for:

https://vaadin.com/roadmap
https://vaadin.com/blog/jakarta-ee-is-becoming-mainstream-get-ready-for-spring-boot-3-and-vaadin-24

Looks like a March 2023 release date.

**Comment by anoblet** [comment]
@saas786 

> Why we still have @cwmr/iron-star-rating as deps? Since https://github.com/conversionxl/aybolit/pull/228 we now have our own custom star icon?

I agree with you, this should have been removed in #228 . I will create a PR to remove it.

There's no up/down side to merging this PR now. It feels like we're kicking the can down the road though :)

**Comment by saas786** [comment]
> @saas786
> 
> > Why we still have @cwmr/iron-star-rating as deps? Since #228 we now have our own custom star icon?
> 
> I agree with you, this should have been removed in #228 . I will create a PR to remove it.

Ok, will approve: https://github.com/conversionxl/aybolit/pull/240

> > I think we should wait for:
> > https://vaadin.com/roadmap
> > https://vaadin.com/blog/jakarta-ee-is-becoming-mainstream-get-ready-for-spring-boot-3-and-vaadin-24
> > Looks like a March 2023 release date.
> 
> There's no up/down side to merging this PR now. It feels like we're kicking the can down the road though :) From a DX standpoint, someone might be more inclined to use `iron-icon` without it.

No actual benefit to use it now, instead will need testing and search & replace chore on production website.
So deferring is better choice. If I didn't saw the `v24` alpha release, I would have inclined to merge this, but seeing that, lets put it on hold.

**Review by lkraav — APPROVED** [review]
Can you rebase to resolve conflicts @anoblet `yarn.lock` needs to be bumped for 23.3.0.

**Review by lkraav — APPROVED** [review]
By manually loading `vaadin-iconset`, we are able to avoid loading `iron-icon` already in Vaadin 23 https://github.com/conversionxl/aybolit/pull/237/files#diff-a345c5d4e6c635e0f1269dc7e0969c1b2bae2b222f29c44a51c730ea93b0c6bfR6-R7

**Review by lkraav — APPROVED** [review]
Figured out details on how to get `<iron-icon>` out, all Storybooks lgtm.

**Pull request merged**
